ngx_http_apicast_module.c

```diff
@@ -178,6 +206,21 @@ ngx_int_t ngx_http_apicast_set_proxy_cert_if_set(
       err = "SSL_USE_certificate failed";
       goto ssl_failed;
   }
+
+  if (ctx->proxy_client_cert_chain == NULL) {
+    // No chain, that it's expected
+    return NGX_OK;
+  }
+
+  ngx_log_error(NGX_LOG_DEBUG, r->connection->log, 0,
+    "SetProxyCert: set proxy certificate chain");
+  rc = SSL_set1_chain(conn->ssl->connection, ctx->proxy_client_cert_chain);
+  if (rc == 0 ) {
+    err = "SSL chain cert failed";
+    sk_X509_pop_free(ctx->proxy_client_cert_chain, X509_free);
+    goto ssl_failed;
+  }
+
   return NGX_OK;
```

Comment:
I'd set a unique exit point for `NGX_OK` to help follow the control flow. In this particular snippet you could test `ctx->proxy_client_cert_chain` and `goto success;` placing `success:` right above `return NGX_OK;`, but I'd suggest you instead test for the case that you have a client cert chain.
```suggestion
  if (ctx->proxy_client_cert_chain) {
    /* got a client cert chain */
    ngx_log_error(NGX_LOG_DEBUG, r->connection->log, 0,
      "SetProxyCert: set proxy certificate chain");
    rc = SSL_set1_chain(conn->ssl->connection, ctx->proxy_client_cert_chain);
    if (rc == 0 ) {
      err = "SSL chain cert failed";
      sk_X509_pop_free(ctx->proxy_client_cert_chain, X509_free);
      goto ssl_failed;
    }
  }
  
  return NGX_OK;
```